### PR TITLE
Redis - Implement the search group

### PR DESCRIPTION
**Pull request by cescoffier**
This groups allows full-text search over hashes and json documents.
The search commands require Redis Search to be installed on the server.

These commands are marked as experimental until we get more feedback.

### CODE REVIEW

File: docs/src/main/asciidoc/redis-reference.adoc [hunk 1]
```diff
@@ -282,7 +282,9 @@ As mentioned above, the API is divided into groups:
 - cuckoo - `.cuckoo()` (requires the https://redis.com/modules/redis-bloom/[redis-bloom] module on the server side, which also provides the cuckoo filter commands)
 - count-min - `.countmin()` (requires the https://redis.com/modules/redis-bloom/[redis-bloom] module on the server side, which also provides the count-min filter commands)
 - top-k - `.topk()` (requires the https://redis.com/modules/redis-bloom/[redis-bloom] module on the server side, which also provides the top-k filter commands)
-- graph - `.graph()` (requires the https://redis.com/modules/redis-graph/[redis-graph] module on the server side, which also provides the graph commands)
+- graph - `.graph()` (requires the https://redis.com/modules/redis-graph/[redis-graph] module on the server side).
+These commands are marked as experimental, as we would need feedback before making them stable.
+- search - `.search()` (requires the https://redis.com/modules/redis-search/[redis-search] module on the server side).
```
Comment (gsmet) [inline docs/src/main/asciidoc/redis-reference.adoc:287]: Very minor but you're writing the module name both `RedisSearch` and `redis-search`. Maybe we should be consistent? (and it's probably valid for the others)

**Review by gsmet — COMMENTED** [review]
I added a very minor comment. Feel free to ignore.

File: docs/src/main/asciidoc/redis-reference.adoc (hunk 1 above)
Comment (cescoffier) [reply docs/src/main/asciidoc/redis-reference.adoc:287]: The module name (code you deploy on the server) is `RedisSearch`, but it's often referred as redis-search... However, in this context, I believe it should be RedisSearch.

Comment (cescoffier) [reply docs/src/main/asciidoc/redis-reference.adoc]: Fixed (as well as for the other modules)

**Comment by geoand** [comment]
@machi1990 can you have a look at this one please?

**Review by machi1990 — APPROVED** [review]
lgtm

Thanks @cescoffier

**Pull request merged**